src/main/java/org/firmata4j/firmata/device/DeviceConnector.java

```diff
@@ -0,0 +1,63 @@
+package org.firmata4j.firmata.device;
+
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * thew hardware than Firmata is implemented on
+ */
+public abstract class DeviceConnector {
+
+    /**
+     * queue of the received messages.
+     * any device will just queue received data here and leave the rest to FirmataDevice
+     */
+    private BlockingQueue<byte[]> byteQueue;
+
+    /**
+     * Device connector need to know where to put received data on.
+     * if must be present before starting the device.
+     *
+     * @param byteQueue byte queue from frimataDevice
+     */
+    public final void setByteQueue(BlockingQueue<byte[]> byteQueue) {
+        this.byteQueue = byteQueue;
+    }
+
+    protected void queueDeviceResponse(byte[] bytes) {
+        if (bytes == null || bytes.length == 0) {
+            return;
+        }
+        while (!byteQueue.offer(bytes)) {
+            // trying to place bytes to queue until it succeeds
+            // in the previous implemetation, the read bytes would get ignored if queue does not have enough space.
+            // it this implementation the order of received messaged my change in cuncurrent message situation.
+            // maybe making this method synchronize? or changing the implementation way ?
+
+        }
+    }
+
+    /**
+     * will start device and communication with it.
+     */
+    public final void startDevice() throws IOException {
```

Comment:
Name of the method got to be alligned to the name of the class. `startConnector` is better name in case the classname is `DeviceConnector`. `startTransport` is for `FirmataTransportInterface`. The same goes to `stop` method.